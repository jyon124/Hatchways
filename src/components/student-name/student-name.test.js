import React from "react";
import { mount } from "enzyme";
import { waitFor } from "@testing-library/react";
import StudentName from "./student-name";

describe("Student Name Component Tests", () => {
    it("Renders a Student Name component", async () => {
        expect.assertions(2);
        const mockStudent = {
            firstName: "Nate",
            lastName: "Won",
        };
        const wrapper = mount(<StudentName student={mockStudent} />);
        await waitFor(() => {
            expect(wrapper.props().student).toBe(mockStudent);
            expect(wrapper.html()).toMatchSnapshot();
        });
    });
    it("Tests if the Student full-name has turned to the upper case letters", async () => {
        expect.assertions(1);
        const mockStudent = {
            firstName: "nate",
            lastName: "won",
        };
        const wrapper = mount(<StudentName student={mockStudent} />);
        await waitFor(() => {
            expect(wrapper.contains("NATE WON")).toEqual(true);
        });
    });
});
