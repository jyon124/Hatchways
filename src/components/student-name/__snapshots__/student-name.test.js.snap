// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Student Name Component Tests Renders a Student Name component 1`] = `"<h1 class=\\"name\\">NATE WON</h1>"`;
